packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2366,8 +2367,12 @@ export class StripeHelper extends StripeHelperBase {
       throw error.unknownCustomer(invoice.customer);
     }
 
+    // Prorations should not be used for plan details since they can belong
+    // to other plans in the case of an upgrade
+    const charges = invoice.lines.data.filter((line) => !line.proration);
```

Comment:
As I mentioned above, `proration: true` invoice line items can be charges, so perhaps this variable can be renamed to e.g. `subscriptionLineItem`, and perhaps we can filter by `line.type === 'subscription'`? We actually do this further down in this method for a similar reason (to get the `nextInvoiceDate`).

Since they use the same logic, perhaps `nextInvoiceDate` can be obtained directly from the `subscriptionLineItem` to avoid the duplicate filtering work.

Reply:
Sounds good.

Is there any potential situation in which there would be more than one subscription line item?

Reply:
Right now we only have one subscription per invoice (so a user who has multiple subscriptions would receive an invoice email for each one), so it isn't a problem. In the future, we may want to support consolidated billing where we only send one invoice for all the subscriptions a user has.

It's worth noting that your previous logic was also taking the first result (with `proration: false`).

Reply:
Cool, thanks for answering that!

> IIRC, your previous logic was also taking the first result (with proration: false).

Yeah, was more asking this since I saw single-subscription invoices were a pre-existing assumption in the code here.